lib/appRelationsDiscovery.js

```diff
@@ -0,0 +1,65 @@
+const {validate} = require('./utils');
+
+const supportedSyntax = ['default','mermaid'];
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+async function getRelations(asyncApiDocs,requestedSyntax = 'default') {
+  if (typeof asyncApiDocs !== 'object') throw new Error('You must pass an array of AsyncApiDocuments on which you wish to discover the relations between');
+  if (!supportedSyntax.includes(requestedSyntax)) return;
+
+  const parsedAsyncApiDocs = await validate(asyncApiDocs);
+  const metrics = new Map();
+  
+  parsedAsyncApiDocs.forEach(doc => {
+    if (doc.hasServers()) {
+      const servers = doc.servers();
+
+      for (const [, credentials] of Object.entries(servers)) {
+        const slug = `${credentials.url()},${credentials.protocol()}`;
+        let relation;
+        if (metrics.has(slug)) {
+          relation = metrics.get(slug);
+        } else {
+          relation = new Map();
```

Comment:
I think that if you don't find given relation in `metrics` Map you should set new relation here, not at the end of function. It's much clearer to read and it's a good pattern not to set every this same item to Map with corresponding key.

```suggestion
          relation = new Map();
          metrics.set(slug,relation);
```

Reply:
Thanks! You cleared my doubt here, I was reassigning the same item to Map at the end of the function as I thought that change in `application` won't change the original application in Map 😞

Reply:
You operate on objects, which are references by default in JS, so your code was good, but better is set only once and then only operate on ref, not set this same value to corresponding key nth times.

Reply:
one more lesson learned today :)